Compile error with git HEAD

<!--Thank you for taking the time to submit an issue!-->

## Background information

Compiling latest code, get errors

### What version of the PMIx Reference Library are you using? <!--(e.g., v1.0, v2.1, git master @ hash, etc.)-->

master @ c43e3bfbecb78deb631361dcee61b0e438918c20

### Describe how PMIx was installed <!--(e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)-->

git clone

### Please describe the system on which you are running

* Operating system/version: Fedora 38 (beta)
* Computer hardware: x86/arm
* Network type: TCP

GCC 13.0.1.  N.B. This is on a test machine with beta-release OS, but hopefully is a heads-up.

-----------------------------

## Details of the problem

```
Making all in util
make[2]: Entering directory '/home/tc/src/pmix/build/src/util'
depbase=`echo pmix_if.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/sh ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../../../pmix-git/src/util -I../../src/include -I../../include   -iquote/home/tc/src/pmix/build -iquote/home/tc/src/pmix/pmix-git -iquote/home/tc/src/pmix/pmix-git/src -iquote/home/tc/src/pmix/build/include -I/home/tc/src/pmix/pmix-git/include    -O3 -DNDEBUG  -finline-functions  -fvisibility=hidden -Wundef -Wno-long-long -Wsign-compare -Wmissing-prototypes -Wstrict-prototypes -Wcomment -Wshadow -Werror-implicit-function-declaration -fno-strict-aliasing -pedantic -Wall -Wextra -Werror -MT pmix_if.lo -MD -MP -MF $depbase.Tpo -c -o pmix_if.lo ../../../pmix-git/src/util/pmix_if.c &&\
mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../../pmix-git/src/util -I../../src/include -I../../include -iquote/home/tc/src/pmix/build -iquote/home/tc/src/pmix/pmix-git -iquote/home/tc/src/pmix/pmix-git/src -iquote/home/tc/src/pmix/build/include -I/home/tc/src/pmix/pmix-git/include -O3 -DNDEBUG -finline-functions -fvisibility=hidden -Wundef -Wno-long-long -Wsign-compare -Wmissing-prototypes -Wstrict-prototypes -Wcomment -Wshadow -Werror-implicit-function-declaration -fno-strict-aliasing -pedantic -Wall -Wextra -Werror -MT pmix_if.lo -MD -MP -MF .deps/pmix_if.Tpo -c ../../../pmix-git/src/util/pmix_if.c  -fPIC -DPIC -o .libs/pmix_if.o
../../../pmix-git/src/util/pmix_if.c: In function 'pmix_ifnametoaddr':
../../../pmix-git/src/util/pmix_if.c:93:13: error: 'strcmp' reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
   93 |         if (strcmp(intf->if_name, if_name) == 0) {
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: note: source object is likely at address zero
../../../pmix-git/src/util/pmix_if.c: In function 'pmix_ifnametoindex':
../../../pmix-git/src/util/pmix_if.c:113:13: error: 'strcmp' reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
  113 |         if (strcmp(intf->if_name, if_name) == 0) {
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: note: source object is likely at address zero
../../../pmix-git/src/util/pmix_if.c: In function 'pmix_ifnametokindex':
../../../pmix-git/src/util/pmix_if.c:132:13: error: 'strcmp' reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
  132 |         if (strcmp(intf->if_name, if_name) == 0) {
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: note: source object is likely at address zero
cc1: all warnings being treated as errors
make[2]: *** [Makefile:889: pmix_if.lo] Error 1
```

--- a/src/util/pmix_if.c
+++ b/src/util/pmix_if.c
@@ -17,7 +17,7 @@
  * Copyright (c) 2015-2016 Research Organization for Information Science
  *                         and Technology (RIST). All rights reserved.
  * Copyright (c) 2016-2020 Intel, Inc.  All rights reserved.
- * Copyright (c) 2021-2022 Nanook Consulting.  All rights reserved.
+ * Copyright (c) 2021-2023 Nanook Consulting.  All rights reserved.
  * $COPYRIGHT$
  *
  * Additional copyrights may follow
@@ -79,26 +79,6 @@
 #    endif
 
 /*
- *  Look for interface by name and returns its address
- *  as a dotted decimal formatted string.
- */
-
-int pmix_ifnametoaddr(const char *if_name, struct sockaddr *addr, int length)
-{
-    pmix_pif_t *intf;
-
-    for (intf = (pmix_pif_t *) pmix_list_get_first(&pmix_if_list);
-         intf != (pmix_pif_t *) pmix_list_get_end(&pmix_if_list);
-         intf = (pmix_pif_t *) pmix_list_get_next(intf)) {
-        if (strcmp(intf->if_name, if_name) == 0) {
-            memcpy(addr, &intf->if_addr, length);
-            return PMIX_SUCCESS;
-        }
-    }
-    return PMIX_ERROR;
-}
-
-/*
  *  Look for interface by name and returns its
  *  corresponding pmix_list index.
  */
@@ -110,8 +90,10 @@ int pmix_ifnametoindex(const char *if_name)
     for (intf = (pmix_pif_t *) pmix_list_get_first(&pmix_if_list);
          intf != (pmix_pif_t *) pmix_list_get_end(&pmix_if_list);
          intf = (pmix_pif_t *) pmix_list_get_next(intf)) {
-        if (strcmp(intf->if_name, if_name) == 0) {
-            return intf->if_index;
+        if (*(intf->if_name) != '\0') { // required by gcc 13.0 - not sure why
+            if (strncmp(intf->if_name, if_name, PMIX_IF_NAMESIZE-1) == 0) {
+                return intf->if_index;
+            }
         }
     }
     return -1;
@@ -129,8 +111,10 @@ int16_t pmix_ifnametokindex(const char *if_name)
     for (intf = (pmix_pif_t *) pmix_list_get_first(&pmix_if_list);
          intf != (pmix_pif_t *) pmix_list_get_end(&pmix_if_list);
          intf = (pmix_pif_t *) pmix_list_get_next(intf)) {
-        if (strcmp(intf->if_name, if_name) == 0) {
-            return intf->if_kernel_index;
+        if (*(intf->if_name) != '\0') { // required by gcc 13.0 - not sure why
+            if (strncmp(intf->if_name, if_name, PMIX_IF_NAMESIZE-1) == 0) {
+                return intf->if_kernel_index;
+            }
         }
     }
     return -1;
@@ -185,6 +169,7 @@ int pmix_ifaddrtoname(const char *if_addr, char *if_name, int length)
         }
         return PMIX_ERR_NOT_FOUND;
     }
+    memset(if_name, 0, length);
 
     for (r = res; r != NULL; r = r->ai_next) {
         for (intf = (pmix_pif_t *) pmix_list_get_first(&pmix_if_list);
@@ -463,6 +448,7 @@ int pmix_ifindextoname(int if_index, char *if_name, int length)
 {
     pmix_pif_t *intf;
 
+    memset(if_name, 0, length);
     for (intf = (pmix_pif_t *) pmix_list_get_first(&pmix_if_list);
          intf != (pmix_pif_t *) pmix_list_get_end(&pmix_if_list);
          intf = (pmix_pif_t *) pmix_list_get_next(intf)) {
@@ -483,6 +469,7 @@ int pmix_ifkindextoname(int if_kindex, char *if_name, int length)
 {
     pmix_pif_t *intf;
 
+    memset(if_name, 0, length);
     for (intf = (pmix_pif_t *) pmix_list_get_first(&pmix_if_list);
          intf != (pmix_pif_t *) pmix_list_get_end(&pmix_if_list);
          intf = (pmix_pif_t *) pmix_list_get_next(intf)) {
@@ -714,22 +701,12 @@ void pmix_ifgetaliases(char ***aliases)
 /* if we don't have struct sockaddr_in, we don't have traditional
    ethernet devices.  Just make everything a no-op error call */
 
-int pmix_ifnametoaddr(const char *if_name, struct sockaddr *if_addr, int size)
-{
-    return PMIX_ERR_NOT_SUPPORTED;
-}
-
 int pmix_ifaddrtoname(const char *if_addr, char *if_name, int size)
 {
     return PMIX_ERR_NOT_SUPPORTED;
 }
 
 int pmix_ifnametoindex(const char *if_name)
-{
-    return PMIX_ERR_NOT_SUPPORTED;
-}
-
-int16_t pmix_ifnametokindex(const char *if_name)
 {
     return PMIX_ERR_NOT_SUPPORTED;
 }

--- a/src/util/pmix_if.h
+++ b/src/util/pmix_if.h
@@ -14,7 +14,7 @@
  * Copyright (c) 2008      Sun Microsystems, Inc.  All rights reserved.
  * Copyright (c) 2013      Cisco Systems, Inc.  All rights reserved.
  * Copyright (c) 2016-2020 Intel, Inc.  All rights reserved.
- * Copyright (c) 2021-2022 Nanook Consulting.  All rights reserved.
+ * Copyright (c) 2021-2023 Nanook Consulting.  All rights reserved.
  * $COPYRIGHT$
  *
  * Additional copyrights may follow
@@ -78,15 +78,6 @@ PMIX_EXPORT extern pmix_list_t pmix_if_list;
 /* global flags */
 PMIX_EXPORT extern bool pmix_if_do_not_resolve;
 PMIX_EXPORT extern bool pmix_if_retain_loopback;
-
-/**
- *  Lookup an interface by name and return its primary address.
- *
- *  @param if_name (IN)   Interface name
- *  @param if_addr (OUT)  Interface address buffer
- *  @param size    (IN)   Interface address buffer size
- */
-PMIX_EXPORT int pmix_ifnametoaddr(const char *if_name, struct sockaddr *if_addr, int size);
 
 /**
  *  Lookup an interface by address and return its name.

